modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderItemLocalServiceImpl.java

```diff
@@ -2623,6 +2640,242 @@ private void _validate(
 		}
 	}
 
+	private String _validateJSON(
+			CPDefinition cpDefinition, CPInstance cpInstance, String json)
+		throws PortalException {
+
+		String sanitizedJSON = json;
+
+		if (cpDefinition != null) {
+			int optionRelCount =
+				_cpDefinitionOptionRelLocalService.
+					getCPDefinitionOptionRelsCount(
+						cpDefinition.getCPDefinitionId());
+
+			if (optionRelCount > 0) {
+				JSONArray optionJSONArray = CPJSONUtil.toJSONArray(json);
+
+				for (CPDefinitionOptionRel cpDefinitionOptionRel :
+						cpDefinition.getCPDefinitionOptionRels()) {
+
+					if (cpDefinitionOptionRel.isSkuContributor()) {
+						if (cpInstance == null) {
+							throw new NoSuchCPInstanceException();
+						}
+
+						JSONArray optionRelJSONArray = CPJSONUtil.toJSONArray(
+							_cpDefinitionOptionRelLocalService.
+								getCPDefinitionOptionRelKeysCPDefinitionOptionValueRelKeys(
+									cpInstance.getCPInstanceId()));
+						boolean jsonOptionExists = false;
+
+						for (int i = 0; i < optionRelJSONArray.length(); i++) {
+							JSONObject instanceJSONObject =
+								optionRelJSONArray.getJSONObject(i);
+
+							if (Objects.equals(
+									instanceJSONObject.get("key"),
+									cpDefinitionOptionRel.getKey())) {
+
+								for (int j = 0; j < optionJSONArray.length();
+									 j++) {
+
+									JSONObject optionJSONObject =
+										optionJSONArray.getJSONObject(j);
+
+									String key = optionJSONObject.getString(
+										"key");
+
+									if (Objects.equals(
+											key,
+											cpDefinitionOptionRel.getKey())) {
+
+										jsonOptionExists = true;
+
+										optionJSONObject.put(
+											"skuOptionName",
+											instanceJSONObject.get(
+												"skuOptionName")
+										).put(
+											"skuOptionValueNames",
+											instanceJSONObject.get(
+												"skuOptionValueNames")
+										).put(
+											"value",
+											instanceJSONObject.get("value")
+										);
+
+										break;
```

Comment:
This always replaces the values of these 4 specific items no matter what since as a SKU contributor it should always match these values to our known "key" value. I've done it this way to simplify the checking process (so we don't have to check if null OR not correct), but if there's a reason this shouldn't be done just let me know.